skbio/stats/tests/test_composition.py

```diff
@@ -269,6 +335,85 @@ def test_centralize(self):
+        result = ancom(pd.DataFrame(self.table1),
+                       pd.Series(self.cats1),
+                       multiple_comparisons_correction=None)
+        W, reject = result['W'],  result['reject']
+        npt.assert_allclose(W.values,
+                            np.array([6, 6, 2, 2, 2, 2, 2]))
+        npt.assert_allclose(reject.values,
+                            np.array([True, True, False, False,
+                                      False, False, False], dtype=bool))
+        result = ancom(self.table1, self.cats1,
+                       multiple_comparisons_correction=None)
+        W, reject = result['W'],  result['reject']
+        npt.assert_allclose(W.values,
+                            np.array([6, 6, 2, 2, 2, 2, 2]))
+        npt.assert_allclose(reject.values,
+                            np.array([True, True, False, False,
+                                      False, False, False], dtype=bool))
+
+        result = ancom(self.table1,
+                       self.cats1,
+                       multiple_comparisons_correction='holm-bonferroni',
+                       significance_test=scipy.stats.mannwhitneyu)
+        W, reject = result['W'],  result['reject']
+        npt.assert_allclose(W.values,
+                            np.array([0, 0, 0, 0, 0, 0, 0]))
+        npt.assert_allclose(reject.values,
+                            np.array([False]*7, dtype=bool))
+        result = ancom(self.table1,
+                       self.cats1,
+                       multiple_comparisons_correction=None,
+                       significance_test=scipy.stats.mannwhitneyu)
+        W, reject = result['W'],  result['reject']
+        npt.assert_allclose(W.values,
+                            np.array([6, 6, 2, 2, 2, 2, 2]))
+        npt.assert_allclose(reject.values,
+                            np.array([True,  True, False, False,
+                                      False, False, False], dtype=bool))
+
+        result = ancom(self.table2,
+                       self.cats2,
+                       multiple_comparisons_correction=None,
+                       significance_test=scipy.stats.mannwhitneyu)
+        W, reject = result['W'],  result['reject']
+        npt.assert_allclose(W.values,
+                            np.array([8, 8, 3, 3,
+                                      8, 3, 3, 3, 3]))
+        npt.assert_allclose(reject.values,
+                            np.array([True, True, False, False,
+                                      True, False, False, False, False],
+                                     dtype=bool))
+
+        result = ancom(self.table3,
+                       self.cats3,
+                       multiple_comparisons_correction=None)
+        W, reject = result['W'],  result['reject']
+        npt.assert_allclose(W.values,
+                            np.array([0]*7))
+        npt.assert_allclose(reject.values,
+                            np.array([False]*7))
+
+    def test_ancom_fail(self):
```

Comment:
We could try inspecting the function signature but that seems overkill. I think having a test checking that an error is raised is fine.